docs/scripts/generate_colab_notebooks.py

```diff
@@ -0,0 +1,176 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Script to generate Colab-compatible notebooks from notebook source files.
+
+This script processes jupytext percent-format Python files and:
+1. Injects Colab-specific setup cells (pip install, API key from secrets)
+2. Injects cells before the "Import the essentials" section
+3. Saves the result as .ipynb files in docs/colab_notebooks
+"""
+
+from __future__ import annotations
+
+import argparse
+from pathlib import Path
+
+import jupytext
+from nbformat import NotebookNode
+from nbformat.v4 import new_code_cell, new_markdown_cell
+
+IMPORT_SECTION_MARKER = "### 📦 Import the essentials"
+
+COLAB_SETUP_MARKDOWN = """\
+### ⚡ Colab Setup
+
+Run the cells below to set up the environment for Google Colab.
+"""
+
+ADDITIONAL_DEPENDENCIES = {
+    "4-providing-images-as-context.py": " pillow",
+}
+
+COLAB_INSTALL_CELL = """\
+# Install data-designer and dependencies
+!pip install -q data-designer{}"""
+
+COLAB_API_KEY_CELL = """\
+# Set up NVIDIA API key from Colab secrets
+import os
+
+from google.colab import userdata
+
+os.environ["NVIDIA_API_KEY"] = userdata.get("NVIDIA_API_KEY")"""
+
+
+def create_colab_setup_cells(additional_dependencies: str) -> list[NotebookNode]:
+    """Create the Colab-specific setup cells to inject before imports."""
+    return [
+        new_markdown_cell(source=COLAB_SETUP_MARKDOWN),
+        new_code_cell(source=COLAB_INSTALL_CELL.format(additional_dependencies)),
+        new_code_cell(source=COLAB_API_KEY_CELL),
+    ]
+
+
+def find_import_section_index(cells: list[NotebookNode]) -> int:
+    """Find the index of the 'Import the essentials' markdown cell."""
```

Comment:
Does this mean we always have to have this markdown to detect the section? Wondering if `from data_designer.essentials import` is a more future-proof check?